### ISSUE TITLE
Should invalid URL halt the scraping process?

### ISSUE THREAD

**Issue by rgaudin** [webrecorder/browsertrix-crawler#654]
In the following run, we see that the crawler crashed parsing an incorrect URL. Given this is not a seed URL, I'd expect a warning and that URL to be skipped. WDYT?

```
[zimit::2024-07-23 06:08:14,480] INFO:Browsertrix crawler: version 1.2.4
[zimit::2024-07-23 06:08:14,498] INFO:Running browsertrix-crawler crawl: crawl --failOnFailedSeed --waitUntil load --depth -1 --timeout 90 --behaviors autoplay,autofetch,siteSpecific --behaviorTimeout 90 --sizeLimit 4294967296 --diskUtilization 90 --timeLimit 7200 --url https://www.worldradiohistory.com/BOOKSHELF-ARH/ --userAgentSuffix zimit.kiwix.org+ contact+zimfarm@kiwix.org --mobileDevice Pixel 2 --cwd /output/.tmpc14lrrs7 --statsFilename /output/crawl.json
{"timestamp":"2024-07-23T06:08:17.715Z","logLevel":"info","context":"general","message":"Browsertrix-Crawler 1.2.4 (with warcio.js 2.2.1)","details":{}}
{"timestamp":"2024-07-23T06:08:17.720Z","logLevel":"info","context":"general","message":"Seeds","details":[{"url":"https://www.worldradiohistory.com/BOOKSHELF-ARH/","scopeType":"prefix","include":["/^https?:\\/\\/www\\.worldradiohistory\\.com\\/BOOKSHELF\\-ARH\\//"],"exclude":[],"allowHash":false,"depth":-1,"sitemap":null,"auth":null,"_authEncoded":null,"maxExtraHops":0,"maxDepth":1000000}]}
{"timestamp":"2024-07-23T06:08:20.599Z","logLevel":"info","context":"worker","message":"Creating 1 workers","details":{}}
{"timestamp":"2024-07-23T06:08:20.602Z","logLevel":"info","context":"worker","message":"Worker starting","details":{"workerid":0}}
[snip]
{"timestamp":"2024-07-23T06:10:33.341Z","logLevel":"info","context":"behavior","message":"Run Script Finished","details":{"frameUrl":"https://www.worldradiohistory.com/BOOKSHELF-ARH/Bookshelf_Audio.htm","page":"https://www.worldradiohistory.com/BOOKSHELF-ARH/Bookshelf_Audio.htm","workerid":0}}
{"timestamp":"2024-07-23T06:10:33.341Z","logLevel":"info","context":"behavior","message":"Behaviors finished","details":{"finished":1,"page":"https://www.worldradiohistory.com/BOOKSHELF-ARH/Bookshelf_Audio.htm","workerid":0}}
{"timestamp":"2024-07-23T06:10:34.344Z","logLevel":"info","context":"pageStatus","message":"Page Finished","details":{"loadState":4,"page":"https://www.worldradiohistory.com/BOOKSHELF-ARH/Bookshelf_Audio.htm","workerid":0}}
{"timestamp":"2024-07-23T06:10:35.222Z","logLevel":"info","context":"worker","message":"Starting page","details":{"workerid":0,"page":"https://www.worldradiohistory.com/BOOKSHELF-ARH/Bookshelf_Bernards_Babani.htm"}}
{"timestamp":"2024-07-23T06:10:35.224Z","logLevel":"info","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":29,"total":1310,"pending":1,"failed":0,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2024-07-23T06:10:34.367Z\",\"extraHops\":0,\"url\":\"https:\\/\\/www.worldradiohistory.com\\/BOOKSHELF-ARH\\/Bookshelf_Bernards_Babani.htm\",\"added\":\"2024-07-23T06:08:25.147Z\",\"depth\":1}"]}}
{"timestamp":"2024-07-23T06:10:35.628Z","logLevel":"info","context":"general","message":"Awaiting page load","details":{"page":"https://www.worldradiohistory.com/BOOKSHELF-ARH/Bookshelf_Bernards_Babani.htm","workerid":0}}
{"timestamp":"2024-07-23T06:10:35.901Z","logLevel":"warn","context":"recorder","message":"Request failed","details":{"url":"http://fonts.googleapis.com/css?family=Droid+Sans|Josefin+Slab:600,600italic,bold,bolditalic|Josefin+Sans:600,600italic,bold,bolditalic|Cherry+Cream+Soda","errorText":"","page":"https://www.worldradiohistory.com/BOOKSHELF-ARH/Bookshelf_Bernards_Babani.htm","workerid":0}}
node:internal/url:797
    this.#updateContext(bindingUrl.parse(input, base));
                                   ^

TypeError: Invalid URL
    at new URL (node:internal/url:797:36)
    at createRequest (file:///app/dist/util/recorder.js:1102:23)
    at Recorder.serializeToWARC (file:///app/dist/util/recorder.js:727:31)
    at Recorder.handleRedirectResponse (file:///app/dist/util/recorder.js:209:14)
    at Recorder.handleRequestWillBeSent (file:///app/dist/util/recorder.js:172:18)
    at file:///app/dist/util/recorder.js:76:62
    at file:///app/node_modules/puppeteer-core/lib/esm/third_party/mitt/mitt.js:36:7
    at Array.map (<anonymous>)
    at Object.emit (file:///app/node_modules/puppeteer-core/lib/esm/third_party/mitt/mitt.js:35:20)
    at CdpCDPSession.emit (file:///app/node_modules/puppeteer-core/lib/esm/puppeteer/common/EventEmitter.js:77:23) {
  code: 'ERR_INVALID_URL',
  input: 'http://www%20.worldradiohistory.com/Bookshelf_Bernards_Babani.htm'
}

Node.js v20.15.0
Traceback (most recent call last):
  File "/usr/bin/zimit", line 8, in <module>
    sys.exit(zimit.zimit())
             ^^^^^^^^^^^^^
  File "/app/zimit/lib/python3.12/site-packages/zimit/zimit.py", line 585, in zimit
    run(sys.argv[1:])
  File "/app/zimit/lib/python3.12/site-packages/zimit/zimit.py", line 480, in run
    raise subprocess.CalledProcessError(crawl.returncode, cmd_args)
subprocess.CalledProcessError: Command '['crawl', '--failOnFailedSeed', '--waitUntil', 'load', '--depth', '-1', '--timeout', '90', '--behaviors', 'autoplay,autofetch,siteSpecific', '--behaviorTimeout', '90', '--sizeLimit', '4294967296', '--diskUtilization', '90', '--timeLimit', '7200', '--url', 'https://www.worldradiohistory.com/BOOKSHELF-ARH/', '--userAgentSuffix', 'zimit.kiwix.org+ contact+zimfarm@kiwix.org', '--mobileDevice', 'Pixel 2', '--cwd', '/output/.tmpc14lrrs7', '--statsFilename', '/output/crawl.json']' returned non-zero exit status 1.
```

**Comment by ikreymer** [comment]
No, it should not cause a crash. Can confirm was able to repro this when starting a crawl from a different page, but not on the exact page, though a bit odd, since it shouldn't have even gotten this far with an invalid URL.

**Comment by ikreymer** [comment]
Looks like it was caused by a 'synthetic' redirect, created by the browser from `http://www%20....` -> `https://www%20....` - even though the original invalid URL request failed it still created the redirect. The fix in #658 will check the redirect URL and reject if it is invalid.
